File: Gamestore.ApplicationServices/Models/CommentListingModel.cs
namespace Gamestore.ApplicationServices.Models;
public class CommentListingModel
{
    public int Id { get; set; }

    public string AuthorName { get; set; }

    public string Body { get; set; }

    public bool CanBeDeletedByUser { get; set; }

    public int? AsReplyTo { get; set; }

    public int? AsQuoteTo { get; set; }

    public DateTime? BanUntil { get; set; }
}

File: Gamestore.AngularClient/src/models/CommentListingModel.ts
import { Data } from "@angular/router";

export interface CommentListingModel {
  id: number;
  authorName: string;
  body: string;
  canBeDeletedByUser: boolean;
  asReplyTo: number;
  asQuoteTo: number;
  commentType: CommentType;
  banUntil: Date;
}

export enum CommentType {
  reply,
  quote,
}
